src/compile/data/source.ts

```diff
@@ -77,6 +77,11 @@ export class SourceNode extends DataFlowNode {
   }
 
   public assemble(): VgData {
+    // remove empty format
+    if (this._data.format && keys(this._data.format).length === 0) {
```

Comment:
Would this cause side effect?

Reply:
On the original data? Hmm, let me check that.

Reply:
No problem. We assign a new object to `this._data`.